ui/src/components/group/GroupReviewTable.js

```diff
@@ -116,51 +132,61 @@ class GroupReviewTable extends React.Component {
-                    delete member.memberFullName;
-                });
-            });
-            group.groupMembers = group.groupMembers.filter((m) => {
-                if (
-                    this.state.deletedMembers.has(m.memberName) ||
-                    this.state.extendedMembers.has(m.memberName)
-                ) {
-                    return m;
+            // show prompt for user to ask for confirmation once the user asked to delete member/s
+
+            if (this.state.deletedMembers.size > 0) {
+                this.setState({ showDeleteConfirmation : true });
+            } else {
+                this.updateReviewGroup();
+            }
+        }
+    }
+
+    updateReviewGroup() {
+        //construct role object from state
+        let group = {
+            name: this.props.groupName,
+        };
+        group.groupMembers = produce(this.props.members, (draft) => {
+            draft.forEach((member) => {
+                if (this.state.deletedMembers.has(member.memberName)) {
+                    member.active = false;
                 }
+                member.expiration = null;
+                delete member.memberFullName;
             });
-            this.props
-                .reviewGroup(
-                    this.props.groupName,
-                    group,
-                    this.state.justification,
-                    this.props._csrf
-                )
-                .then(() => {
-                    this.setState({
-                        submittedReview: true,
-                        errorMessage: null,
-                        justification: '',
-                    });
-                    this.props.onUpdateSuccess(
-                        `Successfully submitted the review for group ${this.props.groupName}`
-                    );
-                    this.loadGroup();
-                })
-                .catch((err) => {
-                    this.setState({
-                        errorMessage: RequestUtils.xhrErrorCheckHelper(err),
-                    });
+        });
+        group.groupMembers = group.groupMembers.filter((m) => {
+            if (
+                this.state.deletedMembers.has(m.memberName) ||
+                this.state.extendedMembers.has(m.memberName)
+            ) {
+                return m;
+            }
+        });
+        this.props
+            .reviewGroup(
+                this.props.groupName,
+                group,
+                this.state.justification,
+                this.props._csrf
+            )
+            .then(() => {
+                this.setState({
+                    submittedReview: true,
+                    errorMessage: null,
+                    justification: '',
                 });
-        }
+                this.props.onUpdateSuccess(
+                    `Successfully submitted the review for group ${this.props.groupName}`
+                );
+                this.loadGroup();
```

Comment:
Is calling loadGroup necessary since we are routing to the members page through onUpdateSuccess